src/map/chat.cpp

```diff
@@ -115,51 +119,47 @@ int chat_createpcchat(map_session_data* sd, const char* title, const char* pass,
 	} else
 		clif_createchat(sd,1);
 
-	return 0;
+	return;
 }
 
  /**
  * Join an existing chat room.
  * @param sd : player requesting
  * @param chatid : ID of the chat room
  * @param pass : password of chat room
- * @return 0
  */
-int chat_joinchat(map_session_data* sd, int chatid, const char* pass)
-{
-	struct chat_data* cd;
+void JoinChat(map_session_data* sd, int chatid, const char* pass){
+	nullpo_retv(sd);
 
-	nullpo_ret(sd);
+	ChatData* cd = (ChatData*)map_id2bl(chatid);
```

Comment:
we should replace every usage of `(ChatData*)map_id2bl(chatid)` with `map_id2cd(chatid)`; we get the BL_CAST for free